xml/ha_virtualization.xml

```diff
@@ -0,0 +1,484 @@
+        </para>
+      </step>
+    </procedure>
+  </sect1>
+
+  <sect1 xml:id="sec-ha-virtualization-adding-virtual-machines">
+    <title>Adding virtual machines as cluster resources</title>
+    <para>
+      Repeat this procedure for each virtual machine you want to create.
+    </para>
+    <procedure xml:id="pro-ha-virtualization-adding-virtual-machines">
+      <title>Adding virtual machines as cluster resources</title>
+      <step>
+        <para>
+          Install a virtual machine on one of the cluster nodes, with the following restrictions:
+        </para>
+        <itemizedlist>
+          <listitem>
+            <para>
+              The installation source and storage must be on shared devices.
+            </para>
+          </listitem>
+          <listitem>
+            <para>
+              Do not configure the VM to start on host boot.
+            </para>
+          </listitem>
+        </itemizedlist>
+        <para>
+          For more information, see
+          <link xlink:href="https://documentation.suse.com/sles/html/SLES-all/book-virtualization.html">
+          <citetitle>&virtual;</citetitle> for &sles;</link>.
+        </para>
+      </step>
+      <step>
+        <para>
+          If the virtual machine is running, shut it down. The cluster will start the VM
+          after you add it as a cluster resource.
+        </para>
+      </step>
+      <step>
+        <para>
+          Dump the virtual machine's XML configuration to a file on the &ocfs; volume:
+        </para>
+<screen>&prompt.root;<command>virsh dumpxml vm1 &gt; /mnt/shared/vm1.xml</command></screen>
+        <important role="compact">
+          <para>
+            Make sure the XML file does not contain any references to unshared local paths.
+          </para>
+        </important>
+      </step>
+      <step>
+        <para>
+          Run <command>crm configure</command> to start the <command>crm</command> interactive shell.
+        </para>
+      </step>
+      <step>
+        <para>
+          Create a primitive resource to manage the virtual machine:
+        </para>
+<screen>&prompt.crm.conf;<command>primitive vm1 VirtualDomain \
+  params config="/mnt/shared/vm1.xml" remoteuri="qemu+ssh://%n/system" \
+  meta allow-migrate=true \
+  op monitor timeout=30s interval=10s \
+  utilization cpu=2 hv_memory=1024</command></screen>
+        <para>
+          The option <literal>allow-migrate=true</literal> enables live migration. If the value is
+          set to <literal>false</literal>, the cluster migrates the VM by shutting it down on
+          one node and restarting it on another node.
+        </para>
+      </step>
+      <step>
+        <para>
+          Create an ordering constraint so that the resources managing the lock files always
+          start before the virtual machine:
+        </para>
+<screen>&prompt.crm.conf;<command>order ord_fs_virt Mandatory: virt-clone vm1</command></screen>
```

Comment:
How would this work with lots of VMs? Do you need to add a new ordering constraint for every VM, or is there a more scalable way to do it? (I'm guessing you wouldn't want to add the VMs to a group since that would make them dependent on each other).

Reply:
Perfect, thanks Roger!

Reply:
Semi-related: the section about resource sets[1] uses curly brackets `{ }` instead of regular parentheses `( )`. Do I need to fix it to use parentheses, or can you use either type? 

[1] https://documentation.suse.com/sle-ha/15-SP5/html/SLE-HA-all/sec-ha-config-basics-constraints.html#sec-ha-config-basics-constraints-rscset-constraints

Reply:
To my knowledge，{ } is the crmsh syntax for "location" constrain and doesn't apply to "order" nor to "coloation".

Reply:
Oh, that's odd that they're different. Thanks, I'll leave that section as-is, then :)